metadata-ingestion/src/datahub/ingestion/source/superset.py

```diff
@@ -288,10 +295,15 @@ def get_platform_from_database_id(self, database_id):
         else:
             platform_name = get_platform_from_sqlalchemy_uri(sqlalchemy_uri)
         if platform_name == "awsathena":
+            self.superset_field_type_mappings = resolve_sql_type(platform="athena")
             return "athena"
         if platform_name == "clickhousedb":
+            self.superset_field_type_mappings = resolve_sql_type(
+                platform="clickhousedb"
+            )
             return "clickhouse"
         if platform_name == "postgresql":
+            self.superset_field_type_mappings = resolve_sql_type(platform="postgres")
```

Comment:
I'm a bit confused by what's going on here

We don't need an instance variable with field type mappings - instead, whenever we need to generate a schemaField, we can just do `SchemaFieldClass(..., type=resolve_sql_type(column_type, platform="athena/postgres/snowflake"))`

Reply:
You're right I don't think we need this to be an instance variable. In superset each dataset could potentially be tied to a different platform/DB. I can look to add a check before SchemaFieldClass
to grab the correct platform then use the `resolve_sql_type`.

Reply:
Not super familiar with Datahub Class typing but I assumed resolve_sql_type will just return the correct class type for me so I can just input it to generate the Schema Field?

I'm getting an error using the resolve_sql_type [method](https://github.com/datahub-project/datahub/pull/11972#discussion_r1866394224), where the SchemaFieldClass seems to be incorrectly constructed: "type": NumberTypeClass({}),

```
def gen_schema_fields(self, column_data: List[Dict[str, str]], db_platform: str) -> List[SchemaField]:
        schema_fields: List[SchemaField] = []
        for col in column_data:
            col_type = col.get("type").lower()
            field = SchemaFieldClass(
                fieldPath=col.get("column_name", ""),
                type=resolve_sql_type(col_type, platform=db_platform),
                nativeDataType="",
                description=col.get("column_name", ""),
                nullable=True,
            )
            schema_fields.append(field)
        return schema_fields
```
```
                                SchemaFieldClass(
                                    {
                                        "fieldPath": "reply_users_count",
                                        "jsonPath": None,
                                        "nullable": True,
                                        "description": "reply_users_count",
                                        "label": None,
                                        "created": None,
                                        "lastModified": None,
                                        "type": NumberTypeClass({}),
                                        "nativeDataType": "",
                                        "recursive": False,
                                        "globalTags": None,
                                        "glossaryTerms": None,
                                        "isPartOfKey": False,
                                        "isPartitioningKey": None,
                                        "jsonProps": None,
                                    }
                                ),
```
I've also tried switching `type=SchemaFieldDataType(type=resolve_sql_type(col_type, platform=db_platform))`
but it also doesn't seem to be working.

Reply:
Here's some example usage: https://github.com/datahub-project/datahub/blob/189f8cefa71b1137801368edf26f633c80599016/metadata-ingestion/src/datahub/ingestion/source/dbt/dbt_common.py#L812

Maybe `SchemaFieldDataType(type=resolve_sql_type(...) or NullType())` might work. mypy should complain if you haven't set it up right

Reply:
returning None instead of NullType allows some other sources to do error reporting based on this

might change that in the future, but probably not right now

Reply:
That makes sense, sounds good